### PR TITLE
[WFCORE-3306] add-user.sh shouldn't throw AddUserFailedException if '%' character is used in user name.

### PR DESCRIPTION
**Pull request by apintea**
https://issues.jboss.org/browse/WFCORE-3306

### CODE REVIEW

File: domain-management/src/main/java/org/jboss/as/domain/management/security/adduser/PromptNewUserState.java [hunk 1]
```diff
@@ -52,6 +52,11 @@ public State execute() {
             theConsole.printf(usernamePrompt);
             String temp = theConsole.readLine(" : ");
             if (temp != null && temp.length() > 0) {
+                // If the username contains '%', insert another '%' to avoid the Formatter Exception (JBEAP-6637)
```
Comment (darranl) [inline domain-management/src/main/java/org/jboss/as/domain/management/security/adduser/PromptNewUserState.java]: This comment should be referencing a WFCORE issue not a JBEAP issue.

**Comment by darranl** [comment]
Can you please squash those two commits into one so all references to JBEAP can be dropped from the history entirely.

**Comment by apintea** [comment]
I have squashed the commits.

"the username with '%' in the name should be verified as being compatible with the Digest authentication mechanism" - I need to write additional tests for the Digests? (DigestMd5MgmtSaslTestCase, DigestSha256MgmtSaslTestCase and DigestSha512MgmtSaslTestCase)

**Comment by bstansberry** [comment]
@darranl ok now?

**Comment by darranl** [comment]
Latest changes look good, can we please at least manually verify Digest authentication accepts these usernames both using the CLI and over HTTP.

**Comment by bstansberry** [comment]
@apintea Please note the request from @darranl above. Someone please comment here when this has been done.

**Comment by darranl** [comment]
@jmesnil I think this is another one we can close, to progress would need some testing that using '%' in the username works correctly - we will likely be revisiting user management fairly soon anyway.

**Pull request closed without merging**